library/model/book.py

```diff
@@ -64,24 +87,35 @@ def can_borrow(self) -> bool:
             raise AttributeError("No such book type...")
 
     def get_approximate_duration(self) -> int:
-        if self._book_type == "Paper":
-            return self.pages * 3 * 60
-        elif self._book_type == "Electronic":
-            return self.pages * 5 * 60
-        elif self._book_type == "Audio":
-            return self.duration
-        else:
-            raise AttributeError("No such book type...")
+        try:
+            return self._duration_by_booktype.get(self._book_type, 0)
```

Comment:
The approx duration is self.pages *  time per page (in mins). This is incorrect.